pkg/apiserver/experiment/experiment.go

```diff
@@ -16,17 +16,40 @@ package experiment
 import (
 	"context"
 	"net/http"
+	"sync"
+	"time"
 
 	"github.com/gin-gonic/gin"
-
 	"github.com/pingcap/chaos-mesh/api/v1alpha1"
 	statuscode "github.com/pingcap/chaos-mesh/pkg/apiserver/status_code"
 	"github.com/pingcap/chaos-mesh/pkg/config"
 	"github.com/pingcap/chaos-mesh/pkg/core"
+	"golang.org/x/sync/errgroup"
 
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var (
+	// TODO(yeya24): don't hardcode this if it is possible
+	kinds = map[string]v1alpha1.ChaosList{
+		v1alpha1.KindPodChaos:     &v1alpha1.PodChaosList{},
+		v1alpha1.KindNetworkChaos: &v1alpha1.NetworkChaosList{},
+		v1alpha1.KindStressChaos:  &v1alpha1.StressChaosList{},
+		v1alpha1.KindIOChaos:      &v1alpha1.IoChaosList{},
+		v1alpha1.KindKernelChaos:  &v1alpha1.KernelChaosList{},
+		v1alpha1.KindTimeChaos:    &v1alpha1.TimeChaosList{},
+	}
+)
+
+// Experiment defines the basic information of an experiment
+type Experiment struct {
```

Comment:
Maybe we also need to add the event field in this struct and store the recent events 

![image](https://user-images.githubusercontent.com/22956341/82319108-98f51780-9a03-11ea-9af3-903ed1039085.png)

Reply:
From this document https://app.swaggerhub.com/apis/ChaosMesh/chaos-server/1.0.0#/experiment/get_experiments_all I don't see the event field. What about updating the APIs first?

Do you think getting the events of one experiment should be part of the experiments list API, or a separate API?

Reply:
At here, we just need the recent events, such as a day, If we use a separate API, the page will pull too much request to chaos-server at the same time.

Reply:
We also can update this in other pr.

Reply:
Then I think two optional parameters `startTime and endTime` is needed for filtering in the frontend.